File: Semestrul 1/Fundamentele Programarii/Sesiune - Practic/repository/repository.py
from domain.tractor import Tractor

class Repository:

    def __init__(self, filename):
        self.__filename = filename

    def load_from_file(self):
        tractoare = []
        with open(self.__filename, 'r') as f:
            lines = f.readlines()
            for line in lines:
                if line != '':
                    line = line.strip()
                    id, denumire, pret, model, data = line.split(";")
                    tractor = Tractor(int(id), denumire, int(pret), model, data)
                    tractoare.append(tractor)
        f.close()
        return tractoare

    def save_to_file(self, tractoare):
        with open(self.__filename, 'w') as f:
            for el in tractoare:
                string = str(el.get_id()) + ';' + str(el.get_denumire()) + ';' + str(el.get_pret()) + ';' + str(el.get_model()) + ';' + str(el.get_data()) + '\n'
                f.write(string)
        f.close()

    def add(self, tractor):
        tractoare = self.load_from_file()
        tractoare.append(tractor)
        self.save_to_file(tractoare)

    def delete(self, id):
        tractoare = []
        for el in self.load_from_file():
            if el.get_id() != id:
                tractoare.append(el)
        self.save_to_file(tractoare)


    def get_all(self):
        return self.load_from_file()

File: Semestrul 2/Programare orientata obiect/Pregatire_Sesiune/Tractoare/repository.cpp
#pragma once

#include <sstream>
#include "repository.h"
#include "fstream"

void Repository::loadFromFile() {
    all.clear();
    ifstream fin(fileName);
    string line;
    while(getline(fin, line)) {
        int id, numarRoti;
        string denumire, tip;
        int nr = 0;
        stringstream linestream(line);
        string curent;
        while(getline(linestream, curent, ',')) {
            if(nr == 0) {
                id = stoi(curent);
            }
            if(nr == 1) {
                denumire = curent;
            }
            if(nr == 2) {
                tip = curent;
            }
            if(nr == 3) {
                numarRoti = stoi(curent);
            }
            nr++;
        }
        all.push_back(Tractor(id, denumire, tip, numarRoti));
    }
    fin.close();
}

void Repository::saveToFile(vector<Tractor> tractoare) {
    ofstream fout(fileName);
    for(auto &t : tractoare) {
        fout<<t.get_id()<<","<<t.get_denumire()<<","<<t.get_tip()<<","<<t.get_numarRoti()<<endl;
    }
    fout.close();
}

void Repository::adaugare(Tractor t) {
    all.push_back(t);
    saveToFile(all);
}
